samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -112,20 +113,30 @@ def prepare(params: dict) -> dict:
     try:
         # initialize terraform application
         LOG.info("Initializing Terraform application")
-        run(["terraform", "init", "-input=false"], check=True, capture_output=True, cwd=terraform_application_dir)
+        invoke_subprocess_with_loading_pattern(
+            command_args={
+                "args": ["terraform", "init", "-input=false"],
+                "cwd": terraform_application_dir,
+                "stdout": PIPE,
+            }
+        )
 
         # get json output of terraform plan
         LOG.info("Creating terraform plan and getting JSON output")
-
         with osutils.tempfile_platform_independent() as temp_file:
-            run(
-                # input false to avoid SAM CLI to stuck in case if the Terraform project expects input, and customer
-                # does not provide it.
-                ["terraform", "plan", "-out", temp_file.name, "-input=false"],
-                check=True,
-                capture_output=True,
-                cwd=terraform_application_dir,
+            process_output = invoke_subprocess_with_loading_pattern(
+                # input false to avoid SAM CLI to stuck in case if the
+                # Terraform project expects input, and customer does not provide it.
+                command_args={
+                    "args": ["terraform", "plan", "-out", temp_file.name, "-input=false"],
+                    "cwd": terraform_application_dir,
+                    "stdout": PIPE,
+                }
             )
+            if isinstance(process_output, bytes):
+                process_output = process_output.decode("utf-8")
+            LOG.debug(process_output)
```

Comment:
won't this LOG all the debug at once? can we stream it instead?

Reply:
I think in this case (debug is enabled)  no idea of having the dots .

Reply:
So you're saying we should stream one or the other? Is there a way of knowing what log level they have configured?